src/Lower.cpp

```diff
@@ -240,6 +245,7 @@ Stmt lower(const vector<Function> &outputs, const string &pipeline_name, const T
         debug(2) << "Lowering after removing varying attributes:\n" << s << "\n\n";
     }
 
+    s = simplify(s);
```

Comment:
added this originally because remove_trivial_for_loops didn't simplify some cases if we didn't call simplify first after the trim_no_ops, but it seems that it is no longer needed now.